crates/cranelift/src/compiler.rs

```diff
@@ -476,39 +476,23 @@ impl wasmtime_environ::Compiler for Compiler {
     }
 }
 
-#[cfg(feature = "incremental-cache")]
-mod incremental_cache {
-    use super::*;
-
-    struct CraneliftCacheStore(Arc<dyn CacheStore>);
-
-    impl cranelift_codegen::incremental_cache::CacheKvStore for CraneliftCacheStore {
-        fn get(&self, key: &[u8]) -> Option<std::borrow::Cow<[u8]>> {
-            self.0.get(key)
-        }
-        fn insert(&mut self, key: &[u8], val: Vec<u8>) {
-            self.0.insert(key, val);
-        }
-    }
+fn compile_maybe_cached<'a>(
+    context: &'a mut Context,
+    isa: &dyn TargetIsa,
+    _cache_ctx: Option<&mut IncrementalCacheContext>,
+) -> Result<(&'a CompiledCode, Vec<u8>), CompileError> {
+    #[cfg(feature = "incremental-cache")]
```

Comment:
I'd prefer that we keep the config options for `incremental-cache` at the function/module level as they are now: as a general principle, we try not to get very fine-grained with conditional compilation directives, because it makes it harder to see exactly what is included in any given configuration ("ifdef hell"). See e.g. the CoW work where (at Alex's direction actually -- I initially did stuff more like this) we have fully separate modules for the CoW and no-CoW cases that provide the same interface. I think we can just revert the one commit in the series that makes this change.

Reply:
Sure, I can do that. But the cache-disabled-at-runtime case is identical to the cache-disabled-at-compile-time case. That means if I want to change the uncached case, I have to change both places and also make sure to build with the incremental-cache feature set both ways. I figured it's better to have the common path be common.

If you still want it this way, I'll factor that code out to a separate function that's called in both implementations.

Reply:
I've removed the "Remove some incremental-cache code duplication" commit and added one to factor out a `compile_uncached` function to use in both configurations. I hope that addresses your concern?

Reply:
Much better -- thanks!